[User][Feature] 정보 입력하기 모달 추가 #862

jusang3057 opened the pull request:
## 개요
* 정보 입력하러 가기 페이지, dialog 생성

## 작업 내용
* lottie가 들어간 Choice Dialog 생성하였습니다.
* lottie가 들어간 정보 입력하러 가기 Activity 생성하였습니다.
* 앱 처음 실행시 & 로그인시 다음 실행
1. 사용자의 정보 일부가 누락되었고 보여진적 없으면 정보 입력하러 가기 Activity 로 이동합니다.
2. 사용자의 정보 일부가 누락되어있지만 한번 보여진적 있으면 Dialog 띄웁니다.

## 추가적인 내용
* SignIn과 Splash에서 이동하는 경로가 너무 많아서 투명한 Activity를 띄워서 Dialog처럼 보이게 하였습니다.
* 좋은 방법은 아닌것 같은데, 더 좋은 방법 있으면 알려주시길 바랍니다.

Commit:
info required modal

Commit:
ktlint

kongwoojin commented on core/designsystem/src/main/java/in/koreatech/koin/core/designsystem/component/dialog/ChoiceDialog.kt:
```diff
@@ -167,15 +172,194 @@ fun ChoiceDialog(
+ * @param negativeButtonColors 부정 버튼 색상
+ * @param lottieRes lottie 파일(json)
+ */
+@OptIn(ExperimentalMaterial3Api::class)
```
디자인 시스템에 들어갈 내용은 아닌 것 같습니다

jusang3057 replied:
lottie를 제외하면 기존의 ChoiceDialog와 동일하여 여기에 넣었습니다.
디자인 시스템이 아니면 어디에 들어가야 할지 잘 모르겠습니다.

kongwoojin replied:
디자인 시스템은 말 그대로 디자인이 정의된 컴포넌트들을 모아 둔 것입니다.
이 디자인이 디자이너에게 컨펌받은 사항도 아니고, 무엇보다 `ChoiceDialog`에서 Lottie만 추가된 것인데, 굳이 디자인 시스템에 정의해야 하는 이유를 모르겠네요

차라리 `user` 모듈이 맞아 보입니다.

kongwoojin commented on core/designsystem/src/main/java/in/koreatech/koin/core/designsystem/component/dialog/ChoiceDialog.kt:
```diff
@@ -167,15 +172,194 @@ fun ChoiceDialog(
+    positiveButtonColors: FilledButtonColors = FilledButtonColors.Primary,
+    negativeButtonColors: OutlinedBoxButtonColors = OutlinedBoxButtonColors.Neutral,
+    @RawRes lottieRes: Int
+//    cancellable: Boolean = true,
```
안 쓰는 parameter는 지워주세요

jusang3057 replied:
기존 ChoiceDialog에 있는 것도 지우겠습니다.

kongwoojin commented on core/designsystem/src/main/java/in/koreatech/koin/core/designsystem/component/dialog/ChoiceDialog.kt:
```diff
@@ -167,15 +172,194 @@ fun ChoiceDialog(
+fun ChoiceDialog(
+    title: @Composable () -> Unit,
+    description: @Composable () -> Unit,
+    onPositive: () -> Unit,
```
`onPositive` 같은 parameter는 default parameter 선언해주세요

jusang3057 replied:
기존 ChoiceDialog에 있는 것도 선언하겠습니다.

kongwoojin commented on core/designsystem/src/main/java/in/koreatech/koin/core/designsystem/theme/ColorPalette.kt:
```diff
@@ -62,7 +62,8 @@ data class KoinColorPalette(
     val info100: Color,
-    val info50: Color
+    val info50: Color,
+    val main500: Color
```
디자인 시스템은 건드리지 말아주세요

kongwoojin commented on core/src/main/res/values/text_appearance.xml:
```diff
@@ -98,4 +98,12 @@
         <item name="lineHeight">19.2sp</item>
     </style>
 
+    <style name="Theme.TransparentActivity" parent="Theme.Material3.DayNight.NoActionBar">
```
배너에서 사용했던 Transparent 테마가 있을 텐데, 확인 해보시길 바랍니다

jusang3057 replied:
이미 사용중인 Transparent 테마는 투명도 50%이고 제가 필요한것은 70%입니다.

kongwoojin replied:
`KAPTheme.Transparent.NoActionBar` 사용하면 될 것 같은데, 확인 부탁드립니다.
지금 만드신 테마도 투명도는 100%입니다.

jusang3057 replied:
테마에서 투명도를 조절할 수 있는것을 몰라서 Activity에서 70%의 투명도를 주고있었습니다. (현제는 테마로 70%주도록 수정중)
기존의 것을 쓰게 되면 투명도 50%를 투명도 70%으로 만들기 위한 추가 투명도가 Activity에 들어가게 되어 가독성이 떨어질듯합니다.

kongwoojin commented on feature/user/build.gradle.kts:
```diff
@@ -22,6 +22,7 @@ android {
 }
 
 dependencies {
+    implementation(libs.lottie.compose.v660)
```
project 다음에 lottie 선언해주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredFullActivity.kt:
```diff
@@ -0,0 +1,37 @@
+class InfoRequiredFullActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val prefs = getSharedPreferences("info_required", Context.MODE_PRIVATE)
```
clean architecture에서 데이터를 presentation 레이어에서 처리하면 안되지 않나요?

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredFullScreen.kt:
```diff
@@ -0,0 +1,134 @@
+    titleText: String,
+    descriptionText: String,
+    modifier: Modifier = Modifier,
+    onPositive: () -> Unit
```
default parameter 선언해주세요

kongwoojin commented on feature/user/src/main/AndroidManifest.xml:
```diff
@@ -71,6 +71,39 @@
                     android:scheme="koin" />
             </intent-filter>
         </activity>
+        <activity
```
하나의 액티비티에서 분기처리는 힘들까요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredFullScreen.kt:
```diff
@@ -0,0 +1,134 @@
+                    start = 24.dp,
+                    end = 24.dp,
+                    top = 16.dp,
+                    bottom = insets.calculateBottomPadding() + 16.dp
```
`insets.calculateBottomPadding()` 대신 windowinsets를 사용하거나 `navigationBarsPadding()`을 사용해주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredFullScreen.kt:
```diff
@@ -0,0 +1,134 @@
+                    bottom = insets.calculateBottomPadding() + 16.dp
+                )
+                .fillMaxWidth()
+                .height(46.dp),
```
버튼 높이 고정하지 말아주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredFullScreen.kt:
```diff
@@ -0,0 +1,134 @@
+                .fillMaxWidth()
+                .height(46.dp),
+            colors = buttonColors,
+            shape = RoundedCornerShape(8.dp),
```
KoinTheme.shapes 사용해주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredModalActivity.kt:
```diff
@@ -0,0 +1,43 @@
+                        val intent = Intent(this@InfoRequiredModalActivity, UserInfoActivity::class.java)
+                        startActivity(intent)
+                        finish()
+                        overridePendingTransition(0, 0) // deprecated : 아직 대체할수 있는 기능이 없음..
```
`overrideActivityTransition` 사용하면 되지 않나요?

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredModalScreen.kt:
```diff
@@ -0,0 +1,72 @@
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color(0f, 0f, 0f, 0.7f)) // 배경 70% 검정 투명
```
KoinTheme.colors.neutral0에 alpha 값을 수정하면 될 것 같네요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredModalScreen.kt:
```diff
@@ -0,0 +1,72 @@
+fun InfoRequiredModalScreen(
+    title: String,
+    description: String,
+    onPositive: () -> Unit,
```
default parameter 선언해주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -20,11 +23,24 @@ import `in`.koreatech.koin.feature.signin.ui.SignInScreen
 @AndroidEntryPoint
 class SignInActivity : ComponentActivity() {
+
+    private lateinit var prefs: SharedPreferences
```
마찬가지입니다.
clean architecture에서 data 처리는 presentation에서 하지 않습니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -20,11 +23,24 @@ import `in`.koreatech.koin.feature.signin.ui.SignInScreen
 class SignInActivity : ComponentActivity() {
+
+    private lateinit var prefs: SharedPreferences
+    private val viewModel: SignInViewModel by viewModels()
```
`SignInScreen` 컴포넌트에서 SignInViewModel을 생성하니, 여기서 또 생성하면 두번 생성됩니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -20,11 +23,24 @@ import `in`.koreatech.koin.feature.signin.ui.SignInScreen
+    private lateinit var prefs: SharedPreferences
+    private val viewModel: SignInViewModel by viewModels()
+    private var isShownBefore: Int = -1
+    private var modal: Boolean = false
```
액티비티가 아니라 Viewmodel로 가야합니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -20,11 +23,24 @@ import `in`.koreatech.koin.feature.signin.ui.SignInScreen
         super.onCreate(savedInstanceState)
+
+        prefs = getSharedPreferences("info_required", MODE_PRIVATE)
+        isShownBefore = prefs.getInt("isShownBefore", -1) // -1:보여지지 않음, 0:보여줬는데 정보 수정 안함, 1:정보수정 완료
```
주석이 없으면 이해하기 힘든 코드가 되겠네요.
`isShownBefore`와 `isInfoFilled`처럼 boolean을 가지는 두개의 preference로 나누면 좋을 것 같네요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -61,22 +93,32 @@ class SignInActivity : ComponentActivity() {
                         data = link.toUri()
                     }
                 startActivity(intent)
+                if (modal) {
```
한번만 추가할 수는 없을까요?

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -18,9 +20,12 @@ import org.orbitmvi.orbit.viewmodel.container
+    private val getUserInfoUseCase: GetUserInfoUseCase
 ) : ViewModel(), ContainerHost<SignInState, SignInSideEffect> {
     override val container = container<SignInState, SignInSideEffect>(SignInState())
+    private var isInfoRequired: Int = 0 // -1 : 정보 완벽함, 0: 로그인 안함, 1 : 정보 없음
```
위와 마찬가지입니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -46,14 +51,47 @@ class SignInViewModel @Inject constructor(
         }
     }
 
+    fun getIsInfoRequired(): Int {
```
굳이 함수로 만들어야 하는건가요?
변수로도 충분할 것 같습니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -46,14 +51,47 @@ class SignInViewModel @Inject constructor(
+                    }.onFailure {
+                        intent {
+                            reduce {
+                                state.copy(loginError = SignInState.LoginError(true, it.message))
```
서버에서 내려준 메시지를 그대로 사용하는건가요?

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -46,14 +51,47 @@ class SignInViewModel @Inject constructor(
+                getUserInfoUseCase()
+                    .onSuccess { user ->
+                        when (user) {
+                            is User.Anonymous -> {
```
`User.Anonymous`와 `User.General`은 합쳐도 될 것 같네요

kongwoojin commented on feature/user/src/main/res/raw/new_koin_lottie.json:
모든 모듈에 lottie 파일을 넣는건 좋은 생각이 아닌 것 같습니다.

kongwoojin commented on gradle/libs.versions.toml:
```diff
@@ -80,6 +81,7 @@ paparazzi = "1.3.0"
 [libraries]
 androidx-junit-ktx = { group = "androidx.test.ext", name = "junit-ktx", version.ref = "junitKtx" }
 androidx-ui-test-junit4-android = { group = "androidx.compose.ui", name = "ui-test-junit4-android", version.ref = "uiTestJunit4Android" }
+lottie-compose-v660 = { module = "com.airbnb.android:lottie-compose", version.ref = "lottieComposeVersion" }
```
라이브러리 이름에 버전명이 들어간 이유가 있을까요?

kongwoojin commented on koin/src/main/java/in/koreatech/koin/ui/main/activity/MainActivity.kt:
```diff
@@ -168,6 +169,7 @@ class MainActivity : KoinNavigationDrawerTimeActivity() {
         viewModel.updateDining()
     }
 
+    @SuppressLint("RestrictedApi")
```
이건 뭔가요?

kongwoojin commented on koin/src/main/java/in/koreatech/koin/ui/splash/SplashActivity.kt:
```diff
@@ -56,10 +59,17 @@ class SplashActivity : ActivityBase() {
     private val splashViewModel by viewModels<SplashViewModel>()
     private val createdTime = System.currentTimeMillis()
 
+    private lateinit var prefs: SharedPreferences
```
여기도 중복되는 코드네요.
중복되는 코드들을 제거할 수는 없을까요?

kongwoojin commented on koin/src/main/java/in/koreatech/koin/ui/splash/viewmodel/SplashViewModel.kt:
```diff
@@ -30,6 +35,17 @@ class SplashViewModel @Inject constructor(
     private val _tokenState = SingleLiveEvent<TokenState>()
     val tokenState: LiveData<TokenState> get() = _tokenState
 
+    var isShownBefore: Int = -1
```
여기도 중복이네요.

kongwoojin reviewed (changes requested):
고생하셨습니다.
다만, clean architecture 관점에서 올바른 코드는 아닌 것 같습니다.
또한, 중복되는 코드들이 많고, 기존의 코드들이 수정 된 부분이 많이 보이는데, 어떻게 하면 중복되는 코드를 줄이고 기존의 코드를 최대한 수정하지 않을 수 있을 지 생각 해보시면 좋을 것 같습니다.

배너가 동일한 방식으로 구현되었으니, 참고 해보시면 될 것 같네요

KYM-P commented on core/designsystem/src/main/java/in/koreatech/koin/core/designsystem/component/dialog/ConfirmDialog.kt:
```diff
@@ -0,0 +1,374 @@
+            description()
+            Spacer(modifier = Modifier.height(24.dp))
+            Row(
+                horizontalArrangement = Arrangement.spacedBy(8.dp)
```
내용이 하나라 spaceBy는 필요 없을 거 같습니다.

KYM-P reviewed (commented):
GOOD (approve 는 아닙니다)

skdud0629 commented on core/designsystem/src/main/java/in/koreatech/koin/core/designsystem/component/dialog/ChoiceDialog.kt:
```diff
@@ -129,8 +135,7 @@ fun ChoiceDialog(
 //    cancellable: Boolean = true,
```
주석 친 부분은 왜 남겨놓은건가요? 필요없다면 지워주세요

skdud0629 commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -61,22 +93,32 @@ class SignInActivity : ComponentActivity() {
                 finish()
             }
         } else {
             if (handleTimetableIntent()) {
```
조건문이 너무 많아서 가독성이 떨어지는데 이런 부분은 else if() 로 표현할 수 없나요?

skdud0629 commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -46,14 +51,47 @@ class SignInViewModel @Inject constructor(
+                            }
+
+                            is User.Student -> {
+                                if (user.name == null || user.phoneNumber == null || user.major == null || user.studentNumber == null) {
```
state에 유효성 검사를 추가하는게 더 깔끔할 것 같습니다

skdud0629 commented on koin/src/main/java/in/koreatech/koin/ui/splash/SplashActivity.kt:
```diff
@@ -138,19 +148,34 @@ class SplashActivity : ActivityBase() {
-    private fun gotoArticleActivityOrDelay() {
+    private fun gotoInfoRequiredActivity() {
+        lifecycleScope.launch {
+            delay()
```
딜레이를 주는 이유가 뭔가요

skdud0629 commented on koin/src/main/java/in/koreatech/koin/ui/splash/SplashActivity.kt:
```diff
@@ -90,8 +100,8 @@ class SplashActivity : ActivityBase() {
             ToastUtil.getInstance().makeShort(R.string.version_check_failed)
```
헬퍼 함수 만들면 좋을 것 같습니다.

skdud0629 reviewed (commented):
수고하셨어요!

Commit:
review 수정

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/repository/ModalRepositoryImpl.kt:
```diff
@@ -0,0 +1,29 @@
+class ModalRepositoryImpl @Inject constructor(
+    context: Context
+) : ModalRepository {
+    private val infoRequiredPrefs: SharedPreferences = context.getSharedPreferences("info_required", Context.MODE_PRIVATE)
```
`DataStore` 사용해주세요

kongwoojin replied:
그리고 상수로 정의해주세요

kongwoojin replied:
근데 여기는 `Repository` 아닌가요
`DataSource`에서 처리하는게 맞아보입니다.

kongwoojin commented on data/src/main/java/in/koreatech/koin/data/repository/ModalRepositoryImpl.kt:
```diff
@@ -0,0 +1,29 @@
+import `in`.koreatech.koin.domain.repository.ModalRepository
+import javax.inject.Inject
+
+class ModalRepositoryImpl @Inject constructor(
```
굳이 `ModalRepository`를 만들어야 하는지 모르겠네요
기존의 `OnBoardingRepository`가 어느정도 용도가 겹치는 것 같고, 유저 관련된 것이니 `UserRepository`에도 넣을 수 있을 것 같네요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredModalScreen.kt:
```diff
@@ -0,0 +1,71 @@
+@Composable
+fun InfoRequiredModalScreen(
+    title: String,
+    description: String,
```
순서가 잘못되었습니다.
일반 parameter -> Modifier -> default parameter가 있는 parameter 순입니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredViewModel.kt:
```diff
@@ -0,0 +1,44 @@
+
+@HiltViewModel
+class InfoRequiredViewModel @Inject constructor(
+    private val modalRepository: ModalRepository
```
UseCase 거쳐주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInActivity.kt:
```diff
@@ -91,18 +111,23 @@ class SignInActivity : ComponentActivity() {
         }
+
+        if (modal) {
+            startActivity(Intent(this, InfoRequiredActivity::class.java).putExtra(InfoRequiredActivity.EXTRA_IS_FULL, false))
```
근데 로그인 직후 메인 액티비티로 넘어갈텐데, 굳이 여기서 한번 더 처리하는 이유가 있나요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInScreen.kt:
```diff
@@ -72,6 +72,7 @@ fun SignInScreen(
             nextRoute = nextRoute
         )
     }
+    viewModel.updateModalInfo()
```
이렇게 하면 Recomposition 때마다 호출됩니다.
`LaunchedEffect` 사용해주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -18,9 +21,18 @@ import org.orbitmvi.orbit.viewmodel.container
+    private val modalRepository: ModalRepository
 ) : ViewModel(), ContainerHost<SignInState, SignInSideEffect> {
     override val container = container<SignInState, SignInSideEffect>(SignInState())
+    var isInfoRequired: Boolean = false
```
위와 마찬가지입니다.
로그인 화면에서 또 확인하는 이유를 모르겠네요

kongwoojin commented on gradle/libs.versions.toml:
```diff
@@ -210,6 +211,7 @@ krossbow-stomp-core = { group = "org.hildan.krossbow", name = "krossbow-stomp-co
 krossbow-websocket-okhttp = { group = "org.hildan.krossbow", name = "krossbow-websocket-okhttp", version.ref = "krossbow" }
 krossbow-stomp-kxserialization = { group = "org.hildan.krossbow", name = "krossbow-stomp-kxserialization", version.ref = "krossbow" }
 krossbow-stomp-kxserialization-json = { group = "org.hildan.krossbow", name = "krossbow-stomp-kxserialization-json", version.ref = "krossbow" }
+play-services-location = { group = "com.google.android.gms", name = "play-services-location", version.ref = "playServicesLocationVersion" }
```
위치 권한을 사용하는 곳이 없을텐데, 추가한 이유가 있을까요?

kongwoojin commented on core/designsystem/src/main/res/raw/new_koin_lottie.json:
왜 디자인 시스템에 있는건가요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredFullScreen.kt:
```diff
@@ -0,0 +1,127 @@
+                    start = 24.dp,
+                    end = 24.dp,
+                    top = 16.dp,
+                    bottom = WindowInsets.navigationBars.asPaddingValues().calculateBottomPadding() + 16.dp
```
modifier에서 `navigationBarsPadding()` 사용해주세요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredModalScreen.kt:
```diff
@@ -0,0 +1,71 @@
+    description: String,
+    onPositive: () -> Unit = {},
+    onNegative: () -> Unit = {},
+    descriptionStyle: TextStyle,
```
default parameter 선언해도 될 것 같네요

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredViewModel.kt:
```diff
@@ -0,0 +1,44 @@
+    private val _event = MutableSharedFlow<UiEvent>()
+    val event = _event.asSharedFlow()
+
+    fun onStart(isFull: Boolean) {
```
함수 이름 수정해주세요
`onStart`만 보고는 무슨 역할을 하는 함수인지 알 수 없습니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/inforequire/InfoRequiredViewModel.kt:
```diff
@@ -0,0 +1,44 @@
+    private val modalRepository: ModalRepository
+) : ViewModel() {
+
+    private val _event = MutableSharedFlow<UiEvent>()
```
굳이 `SharedFlow`를 사용해서 처리해야 하는 부분일까요

kongwoojin commented on koin/src/main/java/in/koreatech/koin/ui/splash/viewmodel/SplashViewModel.kt:
```diff
@@ -30,6 +37,14 @@ class SplashViewModel @Inject constructor(
+    var isInfoRequired: Boolean = false
+    var infoRequiredShown: Boolean = false
+
+    fun updateModalInfo() {
```
`init` 블럭에서 호출하는게 맞을 것 같습니다.

kongwoojin reviewed (changes requested):
고생하셨습니다.
코드 리뷰하면서 몇가지 의문 사항이 있습니다.
1. 배너와 동일한 구조임에도 불구하고, 별도의 스타일을 사용하는게 이해가 안됩니다. 투명도 0.2 차이가 그렇게 중요한 부분인가요
2. 굳이 Splash와 SignIn에서 두번에 걸쳐 구현한 이유를 모르겠습니다. MainActivity 진입시에 확인하면 되지 않나요

추가로, 해결 안된 이슈의 경우, Resolve 처리 하지 말아주세요
이슈 파악하기 힘들어지고 누락될 수 있습니다.

Commit:
review 수정

 - 기존에 있는 Theme사용
 - Designsystem에 있는 ChoiceDialog 복구
 - Splash와 Signin의 Modal기능 Onboarding에 넣음(+ ModalRepo삭제)
 - MutableSharedFlow -> Channel
 - lottie파일 user로 옮김

kongwoojin commented on core/onboarding/src/main/java/in/koreatech/koin/core/onboarding/OnboardingManager.kt:
```diff
@@ -100,6 +107,47 @@ class OnboardingManager @Inject internal constructor(
+                                    if (user.name == null || user.phoneNumber == null || user.major == null || user.studentNumber == null) {
+                                        val check = onboardingRepository.getShouldOnboarding(INFO_REQUIRED_CHECK)
+                                        onboardingRepository.updateShouldOnboarding(INFO_REQUIRED_CHECK, false)
+                                        val intent = Intent(Intent.ACTION_VIEW, "koin://inforequired/activity".toUri())
```
하위 모듈인 core에서 intent를 인지하지 않게 lambda로 action을 받아도 될 것 같네요

kongwoojin commented on core/onboarding/src/main/java/in/koreatech/koin/core/onboarding/OnboardingManager.kt:
```diff
@@ -100,6 +107,47 @@ class OnboardingManager @Inject internal constructor(
+                                        val check = onboardingRepository.getShouldOnboarding(INFO_REQUIRED_CHECK)
+                                        onboardingRepository.updateShouldOnboarding(INFO_REQUIRED_CHECK, false)
+                                        val intent = Intent(Intent.ACTION_VIEW, "koin://inforequired/activity".toUri())
+                                            .putExtra("extra_is_full", check)
```
상수로 수정해주세요

kongwoojin commented on core/onboarding/src/main/java/in/koreatech/koin/core/onboarding/OnboardingManager.kt:
```diff
@@ -100,6 +107,47 @@ class OnboardingManager @Inject internal constructor(
+
+                                        if (check) {
+                                            context.startActivity(intent)
+                                        } else if (ifFailed) {
```
이건 무슨 조건인가요?

jusang3057 replied:
가독성이 떨어져서 바꾸겠습니다.
splash와 signin의 차이때문에 있는 값이였습니다.

jusang3057 replied:
signin은 처음은 풀스크린 띄워주고, 정보를 입력하지 않았을때 다음부터는 modal을 띄워주는데, 
splash는 맨 처음 풀스크린 띄우고 정보 입력하지 않더라도 추가작업이 없습니다.

kongwoojin commented on feature/user/src/main/java/in/koreatech/koin/feature/user/ui/signin/SignInViewModel.kt:
```diff
@@ -21,6 +21,7 @@ class SignInViewModel @Inject constructor(
     private val userLoginUseCase: UserLoginUseCase
 ) : ViewModel(), ContainerHost<SignInState, SignInSideEffect> {
     override val container = container<SignInState, SignInSideEffect>(SignInState())
+    var isSignIn: Boolean = false
```
stateflow 써주세요

kongwoojin replied:
그리고 로그인에 대한 상태를 따로 기록하는 이유가 있나요?

jusang3057 replied:
둘러보기와 로그인 둘다 nextRoute를 사용하기 때문에 구별하였습니다.

kongwoojin commented on gradle/libs.versions.toml:
```diff
@@ -76,6 +76,7 @@ ktlint = "12.1.2"
 junitKtx = "1.2.1"
 uiTestJunit4Android = "1.7.8"
 paparazzi = "1.3.0"
+playServicesLocationVersion = "21.3.0"
```
안 쓰는 라이브러리 버전 지워주세요

kongwoojin commented on koin/src/main/java/in/koreatech/koin/ui/splash/viewmodel/SplashViewModel.kt:
```diff
@@ -19,7 +20,8 @@ import javax.inject.Inject
     private val updateLatestVersionUseCase: UpdateLatestVersionUseCase,
-    private val isTokenSavedInDeviceUseCase: IsTokenSavedInDeviceUseCase
+    private val isTokenSavedInDeviceUseCase: IsTokenSavedInDeviceUseCase,
+    getUserInfoUseCase: GetUserInfoUseCase
```
필요 없는 코드 같네요

kongwoojin reviewed (changes requested):
고생하셨습니다.
코멘트 확인해주세요

Commit:
review 수정

 - showModalIfNeeded에서 Intend제거
 - SignInActivty에서 isSignin 제거
 - 기타 수정

kongwoojin commented on core/designsystem/build.gradle.kts:
```diff
@@ -18,6 +18,7 @@ dependencies {
     implementation(libs.core.ktx)
     implementation(libs.appcompat)
     implementation(libs.material)
+    implementation(libs.lottie.compose)
```
디자인 시스템에는 lottie가 없어도 되지 않나요?

kongwoojin commented on koin/src/main/java/in/koreatech/koin/ui/splash/SplashActivity.kt:
```diff
@@ -138,19 +149,46 @@ class SplashActivity : ActivityBase() {
+        finishWithTransition()
+    }
+
+    @SuppressLint("WrongConstant")
```
이건 뭔가요?

jusang3057 replied:
Activity.OVERRIDE_TRANSITION_OPEN이 들어간 자리는, 원래 특정한 숫자 상수들 중 하나만 들어가야 하는 자리라서 @SuppressLint("WrongConstant") 없이 사용하면 정확한 값 넣어도 Lint가 경고를 띄웁니다.

jusang3057 replied:
@SuppressLint("WrongConstant")없에면 
Must be one of: android.app.Activity.OVERRIDE_TRANSITION_OPEN, android.app.Activity.OVERRIDE_TRANSITION_CLOSE라고 경고 보내는데, 코드에는 android.app.Activity.OVERRIDE_TRANSITION_OPEN가 들어있는 것을 볼 수 있습니다.

kongwoojin replied:
```
private fun finishActivity() {
        finish()
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            overrideActivityTransition(OVERRIDE_TRANSITION_OPEN, 0, 0)
            overrideActivityTransition(OVERRIDE_TRANSITION_CLOSE, 0, 0)
        } else {
            overridePendingTransition(0, 0)
        }
    }
```
배너 코드입니다.
문제가 없는데 왜 여기서는 문제가 되는건지 이해가 안되네요

jusang3057 replied:
문제가 없는데 아래 빨간줄이 나오기 때문에 경고 무시용으로 넣은 것입니다.
배너 코드에서도 올바른 코드이지만 동일한 경고가 발생하고 있습니다.

kongwoojin reviewed (changes requested):
고생하셨습니다

다만 모달 화면에서 그냥 나가버리면 정보를 기입하지 않았어도 다시 창이 뜨지 않네요
수정 해주시면 감사하겠습니다.

Commit:
수정

Commit:
수정

Commit:
review 수정

kongwoojin reviewed (changes requested):
테스트 해봤는데, n회차 때는 모달이 안 뜨네요

kongwoojin reviewed (approved):
고생하셨습니다.
다만, 아직까지 개선의 여지가 있어 보이긴 하지만, 배포르 해야하니 일단 approve 해두겠습니다.

추후 개선해주세요
